Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalPipeline.cpp

```diff
@@ -0,0 +1,4398 @@
+
+                shader->SetMatrix44(pc.fReg, local2ndc);
+            } break;
+
+            case plPipeConst::kCameraToNDC: {
+                hsMatrix44 cam2ndc = IGetCameraToNDC();
+
+                shader->SetMatrix44(pc.fReg, cam2ndc);
+            } break;
+
+            case plPipeConst::kWorldToNDC: {
+                hsMatrix44 cam2ndc = IGetCameraToNDC();
+                hsMatrix44 world2cam = GetViewTransform().GetWorldToCamera();
+
+                hsMatrix44 world2ndc = cam2ndc * world2cam;
+
+                shader->SetMatrix44(pc.fReg, world2ndc);
+            } break;
+
+            case plPipeConst::kLocalToWorld:
+                shader->SetMatrix34(pc.fReg, GetLocalToWorld());
+                break;
+
+            case plPipeConst::kWorldToLocal:
+                shader->SetMatrix34(pc.fReg, GetWorldToLocal());
+                break;
+
+            case plPipeConst::kWorldToCamera: {
+                hsMatrix44 world2cam = GetViewTransform().GetWorldToCamera();
+
+                shader->SetMatrix34(pc.fReg, world2cam);
+            } break;
+
+            case plPipeConst::kCameraToWorld: {
+                hsMatrix44 cam2world = GetViewTransform().GetCameraToWorld();
+
+                shader->SetMatrix34(pc.fReg, cam2world);
+            } break;
+
+            case plPipeConst::kLocalToCamera: {
+                hsMatrix44 world2cam = GetViewTransform().GetWorldToCamera();
+
+                hsMatrix44 local2cam = world2cam * GetLocalToWorld();
+
+                shader->SetMatrix34(pc.fReg, local2cam);
+            } break;
+
+            case plPipeConst::kCameraToLocal: {
+                hsMatrix44 cam2world = GetViewTransform().GetCameraToWorld();
+
+                hsMatrix44 cam2local = GetWorldToLocal() * cam2world;
+
+                shader->SetMatrix34(pc.fReg, cam2local);
+            } break;
+
+            case plPipeConst::kCamPosWorld: {
+                shader->SetVectorW(pc.fReg, GetViewTransform().GetCameraToWorld().GetTranslate(), 1.f);
+            } break;
+
+            case plPipeConst::kCamPosLocal: {
+                hsPoint3 localCam = GetWorldToLocal() * GetViewTransform().GetCameraToWorld().GetTranslate();
+
+                shader->SetVectorW(pc.fReg, localCam, 1.f);
+            } break;
+
+            case plPipeConst::kObjPosWorld: {
+                shader->SetVectorW(pc.fReg, GetLocalToWorld().GetTranslate(), 1.f);
+            } break;
+
+            // UNIMPLEMENTED
+            case plPipeConst::kDirLight1:
+            case plPipeConst::kDirLight2:
+            case plPipeConst::kDirLight3:
+            case plPipeConst::kDirLight4:
+            case plPipeConst::kPointLight1:
+            case plPipeConst::kPointLight2:
+            case plPipeConst::kPointLight3:
+            case plPipeConst::kPointLight4:
+            case plPipeConst::kColorFilter:
+            case plPipeConst::kMaxType:
+                break;
```

Comment:
This should probably assert?

Reply:
I'll add an assert here and keep an eye on it. A little worried that even if Direct3D didn't implement these options in their shaders that something may be encoded that uses one of these zombie options. But I did a quick look through of the code and I didn't see these constants hooked up to anything that looked to be working.